internal/storage/project.go

```diff
@@ -3,6 +3,7 @@ package storage
 // ProjectUploader is the interface for uploading bundled project files, later to be used in the Sauce Cloud.
 type ProjectUploader interface {
 	Upload(name string) (ArtifactMeta, error)
+	Locate(name string) (ArtifactMeta, error)
```

Comment:
Just food for thought (essentially just some boolean version). The current implementation is fine 👍 

```suggestion
	IsCached(name string) (bool, error)
```

or

```suggestion
	IsStored(name string) (bool, error)
```

or 

```suggestion
	Exists(name string) (bool, error)
```

Reply:
My only concern with the `bool` signature function, is that we need another func call to get the actual fileId.
There is no direct link between `filename`/`signature` and `storageId`.

It will be encapsulated in implementation in any case, but not sure it will be more clear using a `Exists` + `GetStorageId`.